WPMedia/fusion-news-theme-blocks
# Asub 8192/captcha2 login

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

- Create a new component to handle reCaptcha V2 for `signup`, `signin`, `magicLink`, and `checkout`

- Add reCaptcha V2 component to login block

## Jira Ticket

https://arcpublishing.atlassian.net/browse/ASUB-8192

## Acceptance Criteria

- Update BotChallengeProtection component in [identity-block](https://github.com/WPMedia/arc-themes-blocks/tree/arc-themes-release-version-2.1.4/blocks/identity-block)

- BotChallengeProtection will accept a challengeIn attribute. This attribute could be one of:
'signup' || ‘signin’ || ‘magiclink’ || ‘checkout’

- IF !!`${challengeIn}Recaptcha && recaptchaScore between 0 - 1 &&!!recaptchaSiteKey
- THEN render reCaptchaV3. reCaptchaV3 should be initialized by passing the  recaptchaSiteKey

- Every time you grab a response token from reCAPTCHA component, store the response in localStorage. 

- The BotChallengeProtection should be part of the Login block, and the reCaptcha token should be sent during the call to Identity.login();
- User should be successfully Logged In when reCaptcha v3 is enabled for signin.

- Write the integration tests required to check login is working as expected when signinRechaptcha (reCaptchaV3) is enabled/disabled.

## Test Steps

1. Checkout this branch `git checkout {branch name}`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block`
3. Enable reCaptcha for login by visiting Subscription admin page of the org and site you want to test in
4. Create a page with `login` block or visit a site with login block
5. You should see the recaptcha box in the login block

## Effect Of Changes

### Before

Login block does not contain recaptcha box even when it's enabled for the org/site

### After

reCaptcha box should be visible if enabled and required to pass before logging in.

## Dependencies or Side Effects

_Examples of dependencies or side effects are:_

- Manifest pr for new block:
- Feature pack pr for local development:
- How to deploy to news: https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138682964/News+Theme+2.0+Migration+Development+Deployment+Notes
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Dependency on another PR that needs to be merged first

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
